### ISSUE TITLE
Does not recurse through directories with python scripts present

### ISSUE THREAD

**Issue by mrivnak** [tbillington/kondo#117]
I have a project where I use python to script some misc tasks like testing. It seems that once a `.py` file is detected it'll stop searching any other subdirectories since it marks the project as Python. This might have to get filed under the discovery rework since it might need project types to be non-exclusive which could take some extra refactoring.

Example: since the project root has a `run.py` file, the `cargo` and `nodejs` projects aren't detected

```plaintext
project/
├── api/
│   ├── src/
│   ├── target/
│   ├── tests/
│   ├── Cargo.toml
│   └── test.py
├── web/
│   ├── node_modules/
│   ├── src/
│   └── package.json
└── run.py
```

**Comment by tbillington** [comment]
You're correct that resolving this falls under the rework described in #111 and is an example of #29.

It does poke at an issue that I've been struggling to come up with a solid answer for.

Given the nested project structure you've described, would you expect/prefer that kondo treat `project/` as a singular project, or as 3 separate projects?

**Comment by mrivnak** [comment]
I think I'd prefer it to be detected as one project of multiple types (so a single delete prompt). But how do we determine that the subfolders are part of a single project? I'd imagine checking for a .git folder is a good option there.

**Comment by tbillington** [comment]
I hadn't thought of .git but it is a strong indicator.
